server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -333,7 +333,7 @@ private Iterable<String> getFallbackNodeIds(ShardRouting shard, AtomicBoolean is
             return () -> {
                 if (shard.primary() && isThrottled.get() == false) {
                     var fallbackNodeIds = allocation.routingNodes().getAllNodeIds();
-                    logger.debug("Shard [{}] assignment is temporary not possible. Falling back to {}", shard.shardId(), fallbackNodeIds);
+                    logger.warn("Shard [{}] assignment is temporary not possible. Falling back to {}", shard.shardId(), fallbackNodeIds);
```

Comment:
I believe it is an indication that something is not working as expected if we are falling back to all node ids and it should be visible in the logs by default.

Reply:
Sounds good, I will take a look if this could be moved to stats